platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/HttpSolrClientFactory.java

```diff
@@ -256,62 +184,137 @@ private static KeyStore getKeyStore(String location, String password) {
       }
     }
   }
 
+  private HttpClientBuilder createHttpBuilder() {
+
+    HttpClientBuilder httpClientBuilder =
+        HttpClients.custom()
+            .setDefaultCookieStore(new BasicCookieStore())
+            .setMaxConnTotal(128)
+            .setMaxConnPerRoute(32);
+
+    if (useTls()) {
+      httpClientBuilder.setSSLSocketFactory(
+          new SSLConnectionSocketFactory(
+              getSslContext(),
+              getSupportedProtocols(),
+              getSupportedCipherSuites(),
+              SSLConnectionSocketFactory.BROWSER_COMPATIBLE_HOSTNAME_VERIFIER));
+    }
+
+    if (useBasicAuth()) {
+      httpClientBuilder.setDefaultCredentialsProvider(getCredentialsProvider());
+      httpClientBuilder.addInterceptorFirst(new PreemptiveAuth(new BasicScheme()));
+    }
+
+    return httpClientBuilder;
+  }
+
+  /**
+   * This method retrieves the system property using elevated privileges. The information returned
+   * by this method is visible to any object that can call the method. The supported cipher suites
+   * do not need this protection and therefore this method is not leaking sensitive information.
+   *
+   * @return supported cipher suites as an array
+   */
+  public static String[] getSupportedCipherSuites() {
+    return commaSeparatedToArray(new HttpSolrClientFactory().getProperty(HTTPS_CIPHER_SUITES));
+  }
+
+  /**
+   * This method retrieves the system property using elevated privileges. The information returned
+   * by this method is visible to any object that can call the method. The supported protocols do
+   * not need this protection and therefore this method is not leaking sensitive information.
+   *
+   * @return supported cipher suites as an array
+   */
+  public static String[] getSupportedProtocols() {
+    return commaSeparatedToArray(new HttpSolrClientFactory().getProperty(HTTPS_PROTOCOLS));
+  }
+
+  private boolean useBasicAuth() {
+    return new Boolean(getProperty(SOLR_USE_BASIC_AUTH));
+  }
+
+  private String getCoreUrl(String coreName) {
+    return getSolrUrl() + "/" + coreName;
+  }
+
+  private String getSolrUrl() {
+    return getProperty(SOLR_HTTP_URL);
+  }
+
+  private CredentialsProvider getCredentialsProvider() {
+    CredentialsProvider provider = new BasicCredentialsProvider();
+    UsernamePasswordCredentials credentials =
+        new UsernamePasswordCredentials(getProperty(SOLR_USERNAME), getProperty(SOLR_PASSWORD));
+    provider.setCredentials(AuthScope.ANY, credentials);
+    return provider;
+  }
+
+  private String getCoreDataDir(String coreName) {
+    return concatenatePaths(getCoreDir(coreName), "data");
+  }
+
+  private String getCoreDir(String coreName) {
+    return concatenatePaths(getProperty(SOLR_DATA_DIR), coreName);
+  }
+
+  private static String concatenatePaths(String first, String more) {
+    return Paths.get(first, more).toString();
+  }
+
+  private String getProperty(String propertyName) {
+    return propertyCache.computeIfAbsent(
```

Comment:
❓ Can any of the cached property values be changed on a running system or would they all require a restart?

Reply:
My understanding is that it requires a restart to change the system properties. However, I decided against making them static because they were not static in code when I started. 

I weighed the voices in my that said "they'll never change w/o restart, make it static", and "you don't know everything, leave it the way it is". 

In the past, I broke things out of ignorance, or through a failure to imagine all consequence, so the second voice won. And I left the properties tied to the instance, not the class.

Reply:
Actually, they don't require a restart. There is services in Karaf to allow changing system properties and I believe we do use those in a few places. Now as for those specific properties, I would expect those to require a restart if they are changed by design.